test/message_change_test.dart

```diff
@@ -59,14 +59,14 @@ void _testChange(Message factory(
     var request = factory(headers: {'header1': 'header value 1'});
     var copy = request.change(headers: {});
 
-    expect(copy.headers, same(request.headers));
```

Comment:
These `same()` checks are important. They test that the headers map isn't copied more than necessary.